src/character.cpp

```diff
@@ -4574,6 +4575,36 @@ void Character::set_sleep_deprivation( int nsleep_deprivation )
                                   nsleep_deprivation ) );
 }
 
+void Character::set_fasting_days( int nfasting_days )
+{
+    fasting_days = nfasting_days;
+}
+
+int Character::get_fasting_days() const
+{
+    return fasting_days;
+}
+
+void Character::update_fasting()
+{
+    const int current_fasting_days = get_fasting_days();
+    const int total_entries = ( as_avatar() )->get_calorie_diary_entries();
+    int fast_total_days = 0;
+    while( fast_total_days < total_entries &&
+           ( as_avatar() )->get_daily_ingested_kcal( fast_total_days ) < 100 ) {
```

Comment:
This doesn't appear to account for whether the character actually needs to eat, or their actual required calorie intake (e.g. for plant mutants).

You'll want to check against character::needs_food() and probably something like last day's calorie expenditure / 10.

Reply:
Thanks, I had meant to use get_bmr() but didn't because it changes with weight. Now a copy of get_bmr() is saved so that a fast isn't broken by weight loss.